Exiled.API/Features/Map.cs

```diff
@@ -229,6 +253,189 @@ public static Room FindParentRoom(GameObject objectInRoom)
+            }
+            */
+
+            var ragdollInfo = new Ragdoll.Info()
+            {
+                ownerHLAPI_id = mirrorOwnerId != null ? mirrorOwnerId : @default.ownerHLAPI_id,
+                PlayerId = playerId,
+                DeathCause = hitInfo != default ? hitInfo : @default.DeathCause,
+                ClassColor = role.classColor,
+                FullName = role.fullName,
+                Nick = victimNick,
+            };
+
+            return SpawnRagdoll(role, ragdollInfo, position, rotation, velocity, allowRecall);
+        }
+
+        /// <summary>
+        /// Optimized method to Spawn a ragdoll on the map.
+        /// Will only allocate the newly created GameObject, requires extra work and pre-loaded base game roles.
+        /// </summary>
+        /// <remarks>
+        /// <list type="number">
+        /// <item>
+        /// <para>
+        /// EXILED already has an internal, default Ragdoll.Info: the use of this
+        /// method to try to optimize a plugin is absolutely optional.
+        /// </para>
+        /// We recommend using: Map.SpawnRagdoll(RoleType roleType, string victimNick, Vector3 position)
+        /// </item>
+        /// <item>
+        /// This method should only ever be used if you're dealing with massive
+        /// server-sided lag.
+        /// </item>
+        /// <item>
+        /// Ragdoll.Info's "ownerID" isn't the SteamID, but the
+        /// <see cref="Dissonance.Integrations.MirrorIgnorance.MirrorIgnorancePlayer"/>'s PlayerId field.
+        /// </item>
+        /// </list>
+        /// </remarks>
+        /// <param name="role">Main game's <see cref="global::Role"/> thad defines the role to spawn a ragdoll.</param>
+        /// <param name="ragdollInfo"><see cref="Ragdoll.Info"/> object containing the ragdoll's info.</param>
+        /// <param name="position">Where the ragdoll will be spawned.</param>
+        /// <param name="rotation">The rotation for the ragdoll.</param>
+        /// <param name="velocity">The initial velocity the ragdoll will have, as if it was exploded.</param>
+        /// <param name="allowRecall">Sets this ragdoll as respawnable by SCP-049.</param>
+        /// <returns>The <see cref="Ragdoll"/> component created.</returns>
+        public static global::Ragdoll SpawnRagdoll(
+                global::Role role,
+                Ragdoll.Info ragdollInfo,
+                Vector3 position,
+                Quaternion rotation = default,
+                Vector3 velocity = default,
+                bool allowRecall = false)
+        {
+            if (role.model_ragdoll == null)
+                return null;
+
+            GameObject gameObject = Object.Instantiate(role.model_ragdoll, position + role.ragdoll_offset.position, Quaternion.Euler(rotation.eulerAngles + role.ragdoll_offset.rotation));
+            Mirror.NetworkServer.Spawn(gameObject);
```

Comment:
Spawn the gameobject after you've assigned variables, don't serialize the variables twice as you call this method optimized

Reply:
An absolute scam on my part, lol. I'll fix everything when I find more time. I only checked this because I was showing a friend that my code is actually just comments, but thanks for the tip, makes sense too.